device_posture_rule.go

```diff
@@ -9,6 +9,139 @@ import (
+	Name          string                         `json:"name,omitempty"`
+	Type          string                         `json:"type,omitempty"`
+	Interval      string                         `json:"interval,omitempty"`
+	Identifier    string                         `json:"identifier,omitempty"`
+	Config        DevicePostureIntegrationConfig `json:"config,omitempty"`
+}
+
+// DevicePostureIntegrationResponse represents the response from the get
+// device posture integrations endpoint.
+type DevicePostureIntegrationResponse struct {
+	Result DevicePostureIntegration `json:"result"`
+	Response
+	ResultInfo `json:"result_info"`
+}
+
+// DevicePostureIntegrationListResponse represents the response from the list
+// device posture integrations endpoint.
+type DevicePostureIntegrationListResponse struct {
+	Result []DevicePostureIntegration `json:"result"`
+	Response
+	ResultInfo `json:"result_info"`
+}
+
+// CreateDevicePostureIntegration creates a device posture integration within an account.
+//
+// API reference: TODO
+func (api *API) CreateDevicePostureIntegration(ctx context.Context, accountID string, integration DevicePostureIntegration) (DevicePostureIntegration, ResultInfo, error) {
+	uri := fmt.Sprintf("/%s/%s/devices/posture/integration", AccountRouteRoot, accountID)
+
+	res, err := api.makeRequestContext(ctx, http.MethodPost, uri, integration)
+	if err != nil {
+		return DevicePostureIntegration{}, ResultInfo{}, err
+	}
+
+	var devicePostureIntegrationResponse DevicePostureIntegrationResponse
+	err = json.Unmarshal(res, &devicePostureIntegrationResponse)
+	if err != nil {
+		return DevicePostureIntegration{}, ResultInfo{}, errors.Wrap(err, errUnmarshalError)
+	}
+
+	return devicePostureIntegrationResponse.Result, devicePostureIntegrationResponse.ResultInfo, nil
+}
+
+// UpdateDevicePostureIntegration updates a device posture integration within an account.
+//
+// API reference: TODO
+func (api *API) UpdateDevicePostureIntegration(ctx context.Context, accountID string, integration DevicePostureIntegration) (DevicePostureIntegration, ResultInfo, error) {
+	uri := fmt.Sprintf("/%s/%s/devices/posture/integration/%s", AccountRouteRoot, accountID, integration.IntegrationID)
+
+	res, err := api.makeRequestContext(ctx, http.MethodPatch, uri, integration)
+	if err != nil {
+		return DevicePostureIntegration{}, ResultInfo{}, err
+	}
+
+	var devicePostureIntegrationResponse DevicePostureIntegrationResponse
+	err = json.Unmarshal(res, &devicePostureIntegrationResponse)
+	if err != nil {
+		return DevicePostureIntegration{}, ResultInfo{}, errors.Wrap(err, errUnmarshalError)
+	}
+
+	return devicePostureIntegrationResponse.Result, devicePostureIntegrationResponse.ResultInfo, nil
+}
+
+// DevicePostureIntegration returns a specific device posture integrations within an account.
+//
+// API reference: TODO
+func (api *API) DevicePostureIntegration(ctx context.Context, accountID, integrationID string) (DevicePostureIntegration, ResultInfo, error) {
```

Comment:
do we need `ResultInfo` as a return value?

Reply:
It's not necessary here.